src/components/EventDetails/EventDetails.js

```diff
@@ -243,14 +249,82 @@ const Divider = styled.View`
 
 
 class EventDetail extends React.Component<Props, State> {
+  timer: ?IntervalID;
+  timeout: ?TimeoutID;
+  cachedTxInfo = {};
+
   state = {
     isReceiveModalVisible: false,
     SWActivationModalVisible: false,
   };
 
-  isReceived = ({ to: address }: Object): boolean => {
+  componentDidMount() {
+    if (this.props.event.type !== TRANSACTION_EVENT) return;
+    const txInfo = this.findTxInfo() || {};
```

Comment:
How about moving `|| {}` into `findTxInfo()` method?

Reply:
We need to know that we can't find transaction info in order to know whether we should use cachedTxInfo.